### PR TITLE
pip freeze post

### PR DESCRIPTION
**Pull request by git-benjamin**
- **init wip**
- **add requirements screenshot**
- **formatting**

### CODE REVIEW

File: _posts/2024-08-01-pip-freeze.markdown [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    Numpy,
+    CI/CD,
+    GitHub
+  ]
+---
+
+
+# The Importance of Using pip freeze for Data Engineers and Analysts
+
+Prior to my career in software consulting, I had the (mis)fortune of working in accounting, finance, and analytics. Using tools like Excel, Python, and SQL, my only goal was to deliver value quickly. Concepts such as version control, repeatable code, and GIT were completely foreign to me—something I’m sure many data analysts and data scientists can relate to.
```
Comment (Hamo69) [inline _posts/2024-08-01-pip-freeze.markdown:20]: me - something

File: _posts/2024-08-01-pip-freeze.markdown [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+**Manual Cleanup**: Over time, your project might evolve, and certain packages may no longer be needed. However, pip freeze will still list these packages unless manually removed. It’s essential to keep track of your dependencies and clean up the ones you no longer use to avoid unnecessary clutter.
+
+**Consider Using pipreqs**: For more refined control over your dependencies, consider using pipreqs. This tool generates a requirements.txt file based on your imports, ensuring you only include the necessary packages. It’s a great way to streamline your dependency management and avoid bloating your project with unused libraries.
+
+# Wrapping Up
+
+pip freeze is a vital tool for data engineers and analysts. It helps maintain a stable and predictable environment, allowing you to focus on building robust data solutions without worrying about unexpected dependency issues. By adopting these best practices, you can ensure that your projects remain consistent, reliable, and easier to manage. So, don't just burn through your data—freeze your requirements and safeguard your workflows against the chaos of untracked dependencies.
```
Comment (Hamo69) [inline _posts/2024-08-01-pip-freeze.markdown:119]: my personal preference is to replace best practice with *good* practice

Comment (Hamo69) [inline _posts/2024-08-01-pip-freeze.markdown:119]: maybe just add some kind of Call To Action... ie. get in touch at some email address or a link to our website etc

**Review by Hamo69 — APPROVED** [review]
LGTM

**Review by andrewmtyler — APPROVED** [review]
Looks good Ben

**Review by Hamo69 — APPROVED** [review]

**Pull request merged**